DHT test_ping_all failing

Our testers report the following:

```python
test_ping_all (ipv8.test.dht.test_discovery.TestDHTDiscoveryCommunity) ... FAIL

======================================================================
FAIL: test_ping_all (ipv8.test.dht.test_discovery.TestDHTDiscoveryCommunity)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\heteoma\AppData\Local\Programs\Python\Python37\lib\site-packages\asynctest\case.py", line 297, in run
    self._run_test_method(testMethod)
  File "C:\Users\heteoma\AppData\Local\Programs\Python\Python37\lib\site-packages\asynctest\case.py", line 352, in _run_test_method
    result = method()
  File "C:\workspace\jenkins2\workspace\ipv8\test_ipv8_PR_unittests_windows\ipv8\test\dht\test_discovery.py", line 110, in test_ping_all
    self.assertEqual(self.pinged, None)
AssertionError: <ipv8.dht.routing.Node object at 0x0000000003E9DF48> != None
```

I can reproduce this on Ubuntu 20.04 (using `nose`):

```python
Failure
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/case.py", line 60, in testPartExecutor
    yield
  File "/home/quinten/.local/lib/python3.8/site-packages/asynctest/case.py", line 297, in run
    self._run_test_method(testMethod)
  File "/home/quinten/.local/lib/python3.8/site-packages/asynctest/case.py", line 352, in _run_test_method
    result = method()
  File "/home/quinten/Documents/py-ipv8/ipv8/test/dht/test_discovery.py", line 110, in test_ping_all
    self.assertEqual(self.pinged, None)
  File "/home/quinten/Downloads/pycharm-community-2019.3.3/plugins/python-ce/helpers/pycharm/teamcity/diff_tools.py", line 38, in _patched_equals
    raise error
Exception:  :: <ipv8.dht.routing.Node object at 0x7f3795fb32b0> != None
-------------------- >> begin captured logging << --------------------
asyncio: DEBUG: Using selector: EpollSelector
asyncio: DEBUG: Using selector: EpollSelector
DHTDiscoveryCommunity: DEBUG: Launching DHTDiscoveryCommunity with prefix b'000240d796da96ef4e58fc34ff6cea856d90d8e642cb'.
DHTDiscoveryCommunity: INFO: DHT community initialized (peer mid b'e82e926a824e926bedc7ab4a369f8c1b4943a288')
DHTDiscoveryCommunity: DEBUG: Launching DHTDiscoveryCommunity with prefix b'000240d796da96ef4e58fc34ff6cea856d90d8e642cb'.
DHTDiscoveryCommunity: INFO: DHT community initialized (peer mid b'495aa3c4ad07b7ea14c5437a325f62427bb7e3ef')
DHTDiscoveryCommunity: DEBUG: Deleting peer Peer<216.167.216.14:5847, SVqjxK0Ht+oUxUN6Ml9iQnu34+8=> (key b'495aa3c4ad07b7ea14c5437a325f62427bb7e3ef')
--------------------- >> end captured logging << ---------------------
```

[A secondary concern is that the unit tests passed with this failure.]

--- a/ipv8/test/dht/test_discovery.py
+++ b/ipv8/test/dht/test_discovery.py
@@ -104,10 +104,10 @@ class TestDHTDiscoveryCommunity(TestBase):
         node1.last_queries[-1] -= 100
         self.nodes[0].overlay.ping_all()
         self.assertNotIn(node1, self.nodes[0].overlay.store[node1.mid])
+        self.assertEqual(self.pinged, None)
 
         self.nodes[0].overlay.store_for_me[node1.mid].append(node1)
         self.nodes[0].overlay.ping_all()
-        self.assertEqual(self.pinged, None)
         self.assertIn(node1.mid, self.nodes[0].overlay.store_for_me)
 
         node1.last_response -= 30

